DOCS-9646 add files to be edited. (#342)
mulesoft/docs-runtime-fabric: modules/ROOT/edits/custom-ingress-configuration.adoc

> = Define a Custom Ingress Configuration

Reviewer: Consider the alternative of titling this different, reserving the imperative "Define" for the subtopic that contains the actual user actions (imperative = steps).
Maybe something like "When to Use a Custom..." or "Benefits of Using a Custom..." or similar?

[NOTE]
====
The tasks outline in this topic must be performed by an IT administrator

> ====

Reviewer: Instead of beginning a topic with a notice format, I suggest working it in to the preliminary text. This avoids having contiguous points of visual emphasis (title and Note) and also avoids self-reference ("this topic"), per our Style Reference (SR).

> Runtime Fabric on Self-Managed Kubernetes enables you to edit a template file that specifies a custom Ingress configuration. Using this file you can define values for annotations, rewrite rules, and the ingressClassName.

Reviewer: ```suggestion
Anypoint Runtime Fabric on Self-Managed Kubernetes enables you, as an IT administrator, to edit a template file that specifies a custom Ingress controller configuration. Using this file, you can define values for annotations, rewrite rules, and specify a value for `ingressClassName`.
```

Reviewer: I suggest that you globally search for "ingress" and verify the format used as the proper name Ingress or the common type, ingress.

The template file is a Kubernetes ConfigMap file. After customizing the template, you can use the `rtfctl apply ingress-configmap` command-line utility to apply the custom Ingress configuration to your Kubernetes cluster.

> If you do not apply a custom ConfigMap file based on a template file, Runtime Fabric will default to a standard ingress resource definition. Depending on your ingress controller, this resource definition may not work. In this case, you must apply a custom ConfigMap using the template.

Reviewer: If "Runtime Fabric" here applies to the Kubernetes version, the short forms for that are either 

Anypoint Runtime Fabric on EKS
Anypoint Runtime Fabric on AKS

Reviewer: Try to stick with present tense and avoid "may," per SR; verify capitalization or not of "ingress":
```suggestion
If you do not apply a custom ConfigMap file based on a template file, Runtime Fabric defaults to a standard ingress resource definition. Depending on your ingress controller, this resource definition might not work. In this case, you must apply a custom ConfigMap using the template.
```

> == Procedure

Reviewer: See previous comment about title.

> 

Reviewer: GLOBALLY consider inserting some useful bit of information between a task title and the first step.
In this case, maybe reaffirm that the user must be an IT admin?

. Copy the following into a text file:
+
----
apiVersion: v1
kind: ConfigMap
metadata:
  name: ingress-config
  namespace: rtf
data:
  ingress.annotations: |
    nginx.ingress.kubernetes.io/rewrite-target: /
    kubernetes.io/ingress.class: <name>
  ingress.spec: |
    ingressClassName: <name>
    rules.http.paths.path: /${APP_PATH}
----

> . Edit the text file as required.

Reviewer: Can we be more specific about "as required"? As required by what, to do what, to achieve what purpose?

+

> All changes should be made under the `data` section of this file. Do not modify any other sections of the template. The configuration options you can customize are:

Reviewer: ```suggestion
Change text only in the `data` section of this file. Do not modify any other sections of the template. The configuration options you can customize are:
```
Do we need both of the first two sentences?

+
[%header%autowidth.spread,cols="a,a"]
|===
| Element | Description

> | `ingress.annotations` | Annotations defined in this section are added to the `annotations` section of every Ingress resource created of a Mule application. Each annotation must be defined on a new line. Annotations defined in the template are default Kubernetes annotations that you can override, if required.

Reviewer: Reworded for parallelism with peer Description entries (which all start with a verb):
```suggestion
| `ingress.annotations` | Specifies default Kubernetes annotations that you can override, if required, and adds them to the `annotations` section of every Mule application Ingress resource. Each annotation requires a new line.
```
...or some other rewording that combines the two definition sentences rather than separating them with the new line requirement.

> |`ingress.spec` | Used to configure the resource `spec` paramaters

Reviewer: Parallelism
```suggestion
|`ingress.spec` | Configures the resource `spec` parameters.
```

| `ingressClassName` (optional) | Specifies the class name used in the Ingress resource created for a Mule application.

> Defining the `ingressClassName` is supported only on Kubernetes version 1.18 and higher. Some ingress controllers require you to define `ingressClassName` using annotations. If you are using Kubernetes version 1.17 or lower, specify `<ingress-class-name>` using ingress.annotations. Use only one of these methods for defining `ingressClassName` based on the version of Kubernetes you are using. Do not use both methods in the same file.

Reviewer: ```suggestion
Defining `ingressClassName` is supported only on Kubernetes version 1.18 and later. Some ingress controllers require you to define `ingressClassName` using annotations. If you are using Kubernetes version 1.17 or earlier, specify `<ingress-class-name>` using ingress.annotations. Use only one of these methods for defining `ingressClassName`, based on the version of Kubernetes you are using. Do not use both methods in the same file.
```

| `rules.http.paths.path` | Specifies the `path` value used in the `http rules` section of the Ingress resource.

Generally, this value corresponds to the public URL of a Mule application shown in the application URL preview. To allow this value to be configured dynamically during deployment, use `/$(APP_PATH)` as a placeholder.
|===

. Run `rtfctl apply ingress-configmap <custom-template-file>`, using the file you just created as the value for `custom-template-file`.

After running this command, `rtfctl` validates the template file you provided and performs the following:

> 

Reviewer: If the following actions are chronological, use a numbered list rather than bullets.
Be sure to insert linebreaks between the first phrase or sentence of a step or list item and its followup text, if any, per the SR.

> * Creates the ConfigMap file in the `rtf` namespace. Any errors are written to stdout.

Reviewer: linebreak:
```suggestion
* Creates the ConfigMap file in the `rtf` namespace. 
+
Writes any errors  to `stdout`.
```

* Schedules a job to validate the ingress resource generated from the ConfigMap in the template file.

> * Waits until the job completes to output the status and relevant logs. `rtfctl` displays the status to stdout.

Reviewer: See SR entry for "completes."
```suggestion
* Waits until the job ends to output the status and relevant logs. 
+
`rtfctl` displays the status to `stdout`.
```

== See Also

* xref:index-self-managed.adoc[Runtime Fabric on Self-Managed Kubernetes]
* xref:install-self-managed.adoc[Install Runtime Fabric on Self-Managed Kubernetes]

> * xref:enable-inbound-traffic-self.adoc[Configure Domains and Inbound Traffic (On Self-Managed Kuberenetes)]

Reviewer: GLOBALLY avoid using punctuation in titles:
```suggestion
* xref:enable-inbound-traffic-self.adoc[Configure Domains and Inbound Traffic on Self-Managed Kubernetes]
```
